data/json/effect_on_condition.json

```diff
@@ -301,5 +302,158 @@
     "recurrence_max": "14 days",
     "condition": { "not": { "one_in_chance": 4 } },
     "effect": [ { "u_adjust_var": "migo_interest", "type": "migo_interest", "context": "interest", "adjustment": 1 } ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_CAUSE_PORTAL_STORM",
+    "recurrence_min": 1,
+    "recurrence_max": 5,
+    "condition": {
+      "not": {
+        "or": [
+          { "is_weather": "portal_storm" },
+          { "u_has_var": "portal_storm_active", "type": "weather", "context": "cause", "value": "true" }
+        ]
+      }
+    },
+    "effect": [
+      { "u_add_var": "portal_storm_active", "type": "weather", "context": "cause", "value": "true" },
+      { "u_add_var": "portal_storm_strength", "type": "weather", "context": "weather_effect", "value": "0" },
+      {
+        "u_adjust_var": "portal_storm_strength",
+        "type": "weather",
+        "context": "weather_effect",
+        "adjustment": { "name": "portal_storm_base_strength", "type": "weather", "context": "weather_effect", "default": 0 }
+      },
+      {
+        "u_adjust_var": "portal_storm_base_strength",
+        "type": "weather",
+        "context": "weather_effect",
+        "adjustment": 1
+      },
+      { "u_message": "Portal storm happening.", "type": "bad" }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "portal_strengthen",
+    "recurrence_min": "30 minutes",
+    "recurrence_max": "1 hours",
+    "condition": { "is_weather": "portal_storm" },
+    "deactivate_condition": { "not": { "is_weather": "portal_storm" } },
+    "effect": [
+      { "u_message": "Portals get stronger.", "type": "bad" },
+      { "u_adjust_var": "portal_storm_strength", "type": "weather", "context": "weather_effect", "adjustment": 1 },
+      { "u_lose_var": "portal_storm_active", "type": "weather", "context": "cause" }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "portal_message",
+    "recurrence_min": 15,
+    "recurrence_max": 60,
+    "condition": { "and": [ { "one_in_chance": 10 }, { "is_weather": "portal_storm" }, "u_is_outside", "u_can_see" ] },
+    "deactivate_condition": { "not": { "is_weather": "portal_storm" } },
+    "effect": [ { "u_message": "PORTAL_STORM", "snippet": true, "type": "bad" } ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "radiation_portal",
+    "recurrence_min": 30,
+    "recurrence_max": 60,
+    "condition": {
+      "and": [
+        { "one_in_chance": 10 },
+        { "is_weather": "portal_storm" },
+        "u_is_outside",
+        {
+          "u_compare_var": "portal_storm_strength",
+          "type": "weather",
+          "context": "weather_effect",
+          "op": ">=",
+          "value": 3
+        }
+      ]
+    },
+    "deactivate_condition": { "not": { "is_weather": "portal_storm" } },
+    "effect": [
+      { "u_message": "A portal opens for an instant and a painful light pours out.", "type": "bad" },
```

Comment:
Do radiation storms ignore clothing with the RAD_PROOF flag? From the description this sounds like radiation puring in through the poral and I would expect as a player that a hazmat suit would protect me, but I am not sure if the code checks for that.

Reply:
Good catch, added a check for that.